M2/Macaulay2/packages/Varieties.m2

```diff
@@ -464,11 +587,17 @@ cohomology(ZZ, ProjectiveVariety, CoherentSheaf) := Module => opts -> (p, X, F)
 -- Module of twisted global sections Γ_*(F)
 -----------------------------------------------------------------------------
 
+-- This is an approximation of Gamma_* F, at least with an inclusion from Gamma_>=0 F
 -- TODO: optimize caching: if HH^0(F>=b) is cached above, does this need to be cached?
 -- TODO: should F>=0 be hardcoded?
+minimalPresentation SheafOfRings  := prune SheafOfRings  := SheafOfRings  => opts -> identity
 minimalPresentation CoherentSheaf := prune CoherentSheaf := CoherentSheaf => opts -> F -> (
     cacheHooks(symbol minimalPresentation, F, (minimalPresentation, CoherentSheaf), (opts, F),
-	(opts, F) -> sheaf(F.variety, HH^0 F(>=0)))) -- this is the default algorithm
+	(opts, F) -> (
+	    -- this is the default algorithm
+	    G := sheaf(F.variety, HH^0 F(>=0));
+	    G.cache.pruningMap = sheafMap F.cache.SaturationMap;
+	    G)))
 
 -----------------------------------------------------------------------------
 -- cotangentSheaf, tangentSheaf, and canonicalBundle
```

Comment:
Generalizing to weighted projective spaces introduces several complications.  For example, does one have to deal with the difference between the class group and the Picard group?  Should one work with the stack or the scheme?